src/components/dataset/query/sidebar/QueryViewSidebarItem.jsx

```diff
@@ -17,60 +35,98 @@ export class QueryViewSidebarItem extends React.PureComponent {
     token: PropTypes.string,
   };
 
-  handleChange = value => {
-    const { column, handleChange, tableName } = this.props;
+  // The 'apply' button should only be enabled when there are new changes to be applied.
+  // Comparing against prevProps and prevState will find if a change has been made.
+  componentDidUpdate(prevProps, prevState) {
+    const { filterData, tableName, column } = this.props;
+    const { filterMap } = this.state;
+    // enable the button when there are unsaved changes
+    if (!_.isEqual(prevState.filterMap, filterMap)) {
+      this.setState({ disableButton: _.isEmpty(filterMap.value) });
+    }
+    // disable the button when filters have just been applied
+    if (!_.isEqual(prevProps.filterData, filterData)) {
+      const filters = _.get(filterData, [tableName, column.name], {});
+      this.setState({ filterMap: filters, disableButton: true });
+    }
+  }
+
+  handleChange = (value) => {
+    const { column } = this.props;
+    const { filterMap } = this.state;
     const type = column.datatype === 'string' ? 'value' : 'range';
-    const nameValueType = {
-      name: column.name,
-      value,
-      type,
-    };
-    handleChange(nameValueType, tableName);
+    const exclude = _.get(filterMap, 'exclude', false);
+    this.setState({ filterMap: { value, type, exclude } });
+  };
+
+  applyFilters = () => {
+    const { column, handleChange, tableName } = this.props;
+    const { filterMap } = this.state;
+    handleChange(column.name, filterMap, tableName);
+  };
+
+  // mark the filter as 'exclude' when the checkbox has been checked
+  toggleExclude = (boxIsChecked) => {
+    const { filterMap } = this.state;
+    this.setState({ filterMap: { ...filterMap, exclude: boxIsChecked } });
   };
 
   render() {
     const {
+      classes,
       column,
       dataset,
-      filterData,
       filterStatement,
       joinStatement,
-      handleFilters,
       tableName,
       token,
     } = this.props;
-    switch (column.datatype) {
```

Comment:
Is there a reason why you're not using the switch statement anymore? To me, the switch statement feels a bit easier to read.

Reply:
I couldn't find a way to conditionally assign a component to `item` using a switch, and then I forgot to look into it further (my bad). I just did a quick search and I found I could do this:
```
    const item = ((datatype) => {
      switch (datatype) {
        case 'string':
          return <CategoryWrapper ... />;
        case 'float':
        case 'integer':
          return <RangeFilter ... />;
        default:
          return <div />;
      }
    })(column.datatype);
```
So I can make that edit since I agree it's easier to read.